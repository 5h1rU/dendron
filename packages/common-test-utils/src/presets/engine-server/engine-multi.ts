import { TestPresetEntry } from "../../utils";
import ENGINE_SINGLE_TEST_PRESET from "./engine-single";

const INIT = {
  WITH_STUBS: new TestPresetEntry({
    label: "with stubs",
    before: ENGINE_SINGLE_TEST_PRESET.INIT.WITH_STUBS.before,
    results: async ({}: {}) => {
      return [];
    },
  }),
  results: ENGINE_SINGLE_TEST_PRESET.INIT.WITH_STUBS.results,
};

const ENGINE_MULTI_TEST_PRESET = {
  INIT,
};

export default ENGINE_MULTI_TEST_PRESET;
