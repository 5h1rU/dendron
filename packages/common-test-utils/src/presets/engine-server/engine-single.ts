import {
  DVault,
  NotePropsDictV2,
  NoteUtilsV2,
  SchemaModuleDictV2,
  SchemaModulePropsV2,
  SchemaUtilsV2,
} from "@dendronhq/common-all";
import { note2File } from "@dendronhq/common-server";
import _ from "lodash";
import { NoteTestUtilsV3 } from "../../noteUtils";
import { TestPresetEntry } from "../../utils";

const SCHEMAS = {
  INIT: {
    ROOT: new TestPresetEntry({
      label: "root",
      results: async ({
        schemas,
        vault,
      }: {
        schemas: SchemaModuleDictV2;
        vault: DVault;
      }) => {
        const schemaModRoot = schemas["root"] as SchemaModulePropsV2;
        return [
          {
            actual: _.trim(SchemaUtilsV2.serializeModuleProps(schemaModRoot)),
            expected: _.trim(`
version: 1
imports: []
schemas:
  - id: root
    children: []
    title: root
    parent: root
            `),
          },
          {
            actual: schemaModRoot.vault.fsPath,
            expected: vault.fsPath,
          },
        ];
      },
    }),
  },
};

const INIT = {
  WITH_BACKLINKS: new TestPresetEntry({
    label: "with backlinks",
    preSetupHook: async ({ vaults }: { vaults: DVault[]; wsRoot: string }) => {
      await NoteTestUtilsV3.createNote({
        fname: "alpha",
        body: "[[beta]]",
        vault: vaults[0],
      });
      await NoteTestUtilsV3.createNote({
        fname: "beta",
        body: "[[alpha]]",
        vault: vaults[0],
      });
    },
    results: async ({ notes }: { notes: NotePropsDictV2 }) => {
      const alpha = NoteUtilsV2.getNotesByFname({ fname: "alpha", notes })[0];
      const beta = NoteUtilsV2.getNotesByFname({ fname: "beta", notes })[0];
      const link = _.find(alpha.links, (l) => l.type === "backlink");
      return [
        {
          actual: alpha.links.length,
          expected: 2,
        },
        {
          actual: beta.links.length,
          expected: 2,
        },
        {
          actual: link?.from,
          expected: { fname: beta.fname, vault: beta.vault },
        },
      ];
    },
  }),
  WITH_BACKLINKS_V2: new TestPresetEntry({
    label: "with backlinks/ multiple links ",
    preSetupHook: async ({ vaults }: { vaults: DVault[]; wsRoot: string }) => {
      await NoteTestUtilsV3.createNote({
        fname: "alpha",
        body: "[[beta]],[[beta]]",
        vault: vaults[0],
      });
      await NoteTestUtilsV3.createNote({
        fname: "beta",
        body: "[[alpha]]",
        vault: vaults[0],
      });
    },
    results: async ({ notes }: { notes: NotePropsDictV2 }) => {
      const alpha = NoteUtilsV2.getNotesByFname({ fname: "alpha", notes })[0];
      const beta = NoteUtilsV2.getNotesByFname({ fname: "beta", notes })[0];
      const alphaLinks = _.filter(alpha.links, (l) => l.type === "backlink");
      const betaLinks = _.filter(beta.links, (l) => l.type === "backlink");
      return [
        {
          actual: alpha.links.length,
          expected: 3,
        },
        {
          actual: beta.links.length,
          expected: 3,
        },
        {
          actual: alphaLinks.length,
          expected: 1,
        },
        {
          actual: betaLinks.length,
          expected: 2,
        },
      ];
    },
  }),
  WITH_STUBS: new TestPresetEntry({
    label: "with stubs",
    before: async ({ vault }: { vault: DVault }) => {
      const createNotes = ({ vault }: { vault: DVault }) => {
        const vaultPath = vault.fsPath;
        return Promise.all([
          note2File(
            NoteUtilsV2.create({ fname: "foo.journal.2020.08.29", vault }),
            vaultPath
          ),
          note2File(
            NoteUtilsV2.create({ fname: "foo.journal.2020.08.30", vault }),
            vaultPath
          ),
          note2File(
            NoteUtilsV2.create({ fname: "foo.journal.2020.08.31", vault }),
            vaultPath
          ),
        ]);
      };
      await createNotes({ vault });
    },
    results: async ({
      notes,
      vault,
    }: {
      notes: NotePropsDictV2;
      vault: DVault;
    }) => {
      const stubNotes = _.filter(notes, { stub: true });
      return [
        {
          actual: _.size(stubNotes),
          expected: 3,
        },
        {
          actual: stubNotes[0].vault,
          expected: vault,
        },
      ];
    },
  }),
};

const ENGINE_SINGLE_TEST_PRESET = {
  NOTES: {
    INIT,
  },
  SCHEMAS,
};

export default ENGINE_SINGLE_TEST_PRESET;
