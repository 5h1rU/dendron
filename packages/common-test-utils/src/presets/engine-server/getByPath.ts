import { NoteUtilsV2 } from "@dendronhq/common-all";
import { TestPresetEntryV4 } from "../../utilsv2";
import { NOTE_PRESETS_V4 } from "../notes";

const NOTES = {
  ROOT: new TestPresetEntryV4(async ({ vaults, engine }) => {
    const vault = vaults[0];
    const root = NoteUtilsV2.getNoteByFnameV4({
      fname: "root",
      notes: engine.notes,
      vault,
    });
    const { data } = await engine.getNoteByPath({ npath: "root", vault });
    return [
      {
        actual: data?.changed,
        expected: [],
      },
      {
        actual: data?.note,
        expected: root,
      },
    ];
  }),
  EXISTING_NOTE: new TestPresetEntryV4(
    async ({ vaults, engine }) => {
      const vault = vaults[0];
      const note = NoteUtilsV2.getNoteByFnameV4({
        fname: "foo",
        notes: engine.notes,
        vault,
      });
      const { data } = await engine.getNoteByPath({ npath: "foo", vault });
      return [
        {
          actual: data?.changed,
          expected: [],
        },
        {
          actual: data?.note,
          expected: note,
        },
      ];
    },
    {
      preSetupHook: async ({ vaults, wsRoot }) => {
        const vault = vaults[0];
        await NOTE_PRESETS_V4.NOTE_SIMPLE.create({ vault, wsRoot });
      },
    }
  ),
  NOTE_WITH_CAPS_AND_SPACES: new TestPresetEntryV4(
    async ({ vaults, engine }) => {
      const vault = vaults[0];
      const note = NoteUtilsV2.getNoteByFnameV4({
        fname: "000 Index",
        notes: engine.notes,
        vault,
      });
      const { data } = await engine.getNoteByPath({
        npath: "000 Index",
        vault,
      });
      return [
        {
          actual: data?.changed,
          expected: [],
        },
        {
          actual: data?.note,
          expected: note,
        },
      ];
    },
    {
      preSetupHook: async ({ vaults, wsRoot }) => {
        const vault = vaults[0];
        await NOTE_PRESETS_V4.NOTE_WITH_CAPS_AND_SPACE.create({
          vault,
          wsRoot,
        });
      },
    }
  ),
  NEW_NOTE: new TestPresetEntryV4(async ({ vaults, engine }) => {
    const vault = vaults[0];
    const { data } = await engine.getNoteByPath({
      npath: "bar",
      vault,
      createIfNew: true,
    });
    return [
      {
        actual: data?.changed.map((ent) => ({
          fname: ent.note.fname,
          status: ent.status,
        })),
        expected: [
          { fname: "root", status: "update" },
          { fname: "bar", status: "create" },
        ],
      },
      {
        actual: data?.note?.fname,
        expected: "bar",
      },
    ];
  }),
};
export const ENGINE_GET_NOTE_BY_PATH_PRESETS = {
  NOTES,
};
