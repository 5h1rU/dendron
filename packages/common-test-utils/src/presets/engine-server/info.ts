import _ from "lodash";
import { TestPresetEntryV4 } from "../../utilsv2";
import { NOTE_PRESETS_V4 } from "../notes";

const NOTES = {
  BASIC: new TestPresetEntryV4(
    async ({ engine }) => {
      const info = await engine.info();
      if (info.error) {
        throw Error();
      }
      return [
        {
          actual: _.isEmpty(info.data!.version),
          expected: false,
        },
      ];
    },
    {
      preSetupHook: async ({ vaults, wsRoot }) => {
        await NOTE_PRESETS_V4.NOTE_SIMPLE.create({ wsRoot, vault: vaults[0] });
      },
    }
  ),
};

export const ENGINE_INFO_PRESETS = {
  NOTES,
};
