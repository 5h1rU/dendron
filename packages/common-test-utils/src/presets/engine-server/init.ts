import { NotePropsV2, NoteUtilsV2 } from "@dendronhq/common-all";
import { FileTestUtils } from "../../fileUtils";
import { TestPresetEntryV4 } from "../../utilsv2";
import { NOTE_PRESETS_V4 } from "../notes";

const NOTES = {
  DOMAIN_STUB: new TestPresetEntryV4(
    async ({ wsRoot, vaults, engine }) => {
      const noteRoot = NoteUtilsV2.getNoteByFnameV4({
        fname: "root",
        notes: engine.notes,
        vault: vaults[0],
      }) as NotePropsV2;

      const noteChild = NoteUtilsV2.getNoteByFnameV4({
        fname: "foo",
        notes: engine.notes,
        vault: vaults[0],
      }) as NotePropsV2;
      const checkVault = await FileTestUtils.assertInVault({
        wsRoot,
        vault: vaults[0],
        match: ["foo.ch1.md"],
        nomatch: ["foo.md"],
      });
      return [
        {
          actual: noteRoot.children,
          expected: [noteChild.id],
        },
        {
          actual: checkVault,
          expected: true,
        },
      ];
    },
    {
      preSetupHook: async ({ vaults, wsRoot }) => {
        await NOTE_PRESETS_V4.NOTE_SIMPLE_CHILD({ wsRoot, vault: vaults[0] });
      },
    }
  ),
};
export const ENGINE_INIT_PRESETS = {
  NOTES,
};
