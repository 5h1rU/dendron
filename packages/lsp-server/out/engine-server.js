"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DendronEngineServer = void 0;
const engine_server_1 = require("@dendronhq/engine-server");
class DendronEngineServer {
    constructor(opts) {
        this.opts = opts;
    }
    static getOrCreate(opts) {
        if (!DendronEngineServer._instance) {
            DendronEngineServer._instance = new DendronEngineServer(opts);
        }
        return DendronEngineServer._instance;
    }
    init() {
        const root = this.opts.roots[0];
        if (!root) {
            return;
        }
        const engine = engine_server_1.DendronEngine.getOrCreateEngine({
            root: root.uri,
            forceNew: true,
        });
        this._engine = engine;
        this._engine.init();
    }
}
exports.DendronEngineServer = DendronEngineServer;
//# sourceMappingURL=engine-server.js.map