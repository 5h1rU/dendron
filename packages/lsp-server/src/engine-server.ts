import { WorkspaceFolder } from "vscode-languageserver";
import { DendronEngine } from "@dendronhq/engine-server";

type DendronEngineServerOpts = {
  roots: WorkspaceFolder[];
};

export class DendronEngineServer {
  protected _engine?: DendronEngine;

  constructor(public opts: DendronEngineServerOpts) {}

  static _instance: DendronEngineServer | null;

  static getOrCreate(opts: DendronEngineServerOpts) {
    if (!DendronEngineServer._instance) {
      DendronEngineServer._instance = new DendronEngineServer(opts);
    }
    return DendronEngineServer._instance;
  }

  init() {
    const root = this.opts.roots[0];
    if (!root) {
      return;
    }
    const engine = DendronEngine.getOrCreateEngine({
      root: root.uri,
      forceNew: true,
    }) as DendronEngine;
    this._engine = engine;
    this._engine.init();
  }
}
