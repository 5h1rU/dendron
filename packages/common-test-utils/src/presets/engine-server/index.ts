import { ENGINE_DELETE_PRESETS } from "./delete";
import ENGINE_MULTI_TEST_PRESET from "./engine-multi";
import ENGINE_SINGLE_TEST_PRESET from "./engine-single";
import { ENGINE_GET_NOTE_BY_PATH_PRESETS } from "./getByPath";
import { ENGINE_INIT_PRESETS } from "./init";
import NOTE_REF from "./note-refs";
import { ENGINE_QUERY_PRESETS } from "./query";
import { ENGINE_RENAME_PRESETS } from "./rename";
import { ENGINE_UPDATE_PRESETS } from "./update";
import { ENGINE_WRITE_PRESETS } from "./write";

export const ENGINE_SERVER = {
  NOTE_REF,
  ENGINE_MULTI_TEST_PRESET,
  ENGINE_SINGLE_TEST_PRESET,
  ENGINE_WRITE_PRESETS,
  ENGINE_INIT_PRESETS,
  ENGINE_UPDATE_PRESETS,
  ENGINE_DELETE_PRESETS,
  ENGINE_GET_NOTE_BY_PATH_PRESETS,
  ENGINE_RENAME_PRESETS,
  ENGINE_QUERY_PRESETS,
};

export const ENGINE_PRESETS = [
  // {name: "init", presets: ENGINE_SERVER.ENGINE_INIT_PRESETS},
  // {name: "delete", presets: ENGINE_SERVER.ENGINE_DELETE_PRESETS},
  // {name: "getByPath", presets: ENGINE_SERVER.ENGINE_GET_NOTE_BY_PATH_PRESETS},
  { name: "query", presets: ENGINE_SERVER.ENGINE_QUERY_PRESETS },
  // {name: "rename", presets: ENGINE_SERVER.ENGINE_RENAME_PRESETS},
  // {name: "update", presets: ENGINE_SERVER.ENGINE_UPDATE_PRESETS},
  // {name: "write", presets: ENGINE_SERVER.ENGINE_WRITE_PRESETS},
];
