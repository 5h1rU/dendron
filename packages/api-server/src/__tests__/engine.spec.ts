import _ from "lodash";
import {
  CreateEngineFunction,
  ENGINE_CONFIG_PRESETS,
  ENGINE_PRESETS,
  runEngineTestV4,
} from "@dendronhq/common-test-utils";
import { DendronEngineClient } from "@dendronhq/engine-server";

const createEngine: CreateEngineFunction = ({ wsRoot, vaults }) => {
  return DendronEngineClient.create({
    port: "3005",
    ws: wsRoot,
    vaults: vaults.map((ent) => ent.fsPath),
  });
};

describe("engine, schemas/", () => {
  const nodeType = "SCHEMAS";

  ENGINE_PRESETS.forEach((pre) => {
    const { name, presets } = pre;
    describe(name, () => {
      test.each(
        // @ts-ignore
        _.map(presets[nodeType], (v, k) => {
          return [k, v];
        })
      )("%p", async (_key, TestCase) => {
        const { testFunc, ...opts } = TestCase;
        await runEngineTestV4(testFunc, { ...opts, createEngine, expect });
      });
    });
  });
});

describe("engine, notes/", () => {
  const nodeType = "NOTES";

  ENGINE_PRESETS.forEach((pre) => {
    const { name, presets } = pre;
    describe(name, () => {
      test.each(
        _.map(presets[nodeType], (v, k) => {
          return [k, v];
        })
      )("%p", async (_key, TestCase) => {
        // @ts-ignore
        const { testFunc, ...opts } = TestCase;
        await runEngineTestV4(testFunc, { ...opts, createEngine, expect });
      });
    });
  });
});

describe("engine, config/", () => {
  _.map(ENGINE_CONFIG_PRESETS, (presets, name) => {
    describe(name, () => {
      test.each(
        _.map(presets, (v, k) => {
          return [k, v];
        })
      )("%p", async (_key, TestCase) => {
        // @ts-ignore
        const { testFunc, ...opts } = TestCase;
        await runEngineTestV4(testFunc, { ...opts, createEngine, expect });
      });
    });
  });
});
