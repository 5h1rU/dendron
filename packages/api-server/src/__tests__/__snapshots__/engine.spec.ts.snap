// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`notes delete note w/children 1`] = `
Array [
  "foo.ch1.md",
  "foo.schema.yml",
  "root.md",
  "root.schema.yml",
]
`;

exports[`notes getNoteByPath get root 1`] = `
Object {
  "data": Object {
    "changed": Array [],
    "note": Object {
      "body": "
",
      "children": Array [
        "foo",
      ],
      "created": "1",
      "custom": Object {},
      "data": Object {},
      "desc": "",
      "fname": "root",
      "id": "f6f3f49d-d3f9-465a-a4f2-b3f3aa93cab3",
      "links": Array [],
      "parent": null,
      "title": "Root",
      "type": "note",
      "updated": "1",
      "vault": Object {
        "fsPath": "/var/folders/65/c9lrx0tx6k7b_zlb4v1bs2rr0000gn/T/tmp-59344-rVOoNECcwMwB/vault",
      },
    },
  },
  "error": null,
}
`;

exports[`notes query query child note with schema 1`] = `
Object {
  "data": Array [
    Object {
      "body": "
",
      "children": Array [],
      "created": "1",
      "custom": Object {},
      "data": Object {},
      "desc": "",
      "fname": "foo.ch1",
      "id": "foo.ch1",
      "links": Array [],
      "parent": "foo",
      "schema": Object {
        "moduleId": "foo",
        "schemaId": "ch1",
      },
      "title": "Ch1",
      "type": "note",
      "updated": "1",
      "vault": Object {
        "fsPath": "/var/folders/65/c9lrx0tx6k7b_zlb4v1bs2rr0000gn/T/tmp-59344-7YL5msTp1keX/vault",
      },
    },
    Object {
      "body": "
",
      "children": Array [
        "foo.ch1",
      ],
      "created": "1",
      "custom": Object {},
      "data": Object {},
      "desc": "",
      "fname": "foo",
      "id": "foo",
      "links": Array [],
      "parent": "35de83a0-d5f3-4591-b912-aa6fe553c18c",
      "schema": Object {
        "moduleId": "foo",
        "schemaId": "foo",
      },
      "title": "Foo",
      "type": "note",
      "updated": "1",
      "vault": Object {
        "fsPath": "/var/folders/65/c9lrx0tx6k7b_zlb4v1bs2rr0000gn/T/tmp-59344-7YL5msTp1keX/vault",
      },
    },
  ],
  "error": null,
}
`;

exports[`notes query query root note with schema 1`] = `
Object {
  "data": Array [
    Object {
      "body": "
",
      "children": Array [
        "foo.ch1",
      ],
      "created": "1",
      "custom": Object {},
      "data": Object {},
      "desc": "",
      "fname": "foo",
      "id": "foo",
      "links": Array [],
      "parent": "0a128e8c-bba2-440e-92e8-89a311a17512",
      "schema": Object {
        "moduleId": "foo",
        "schemaId": "foo",
      },
      "title": "Foo",
      "type": "note",
      "updated": "1",
      "vault": Object {
        "fsPath": "/var/folders/65/c9lrx0tx6k7b_zlb4v1bs2rr0000gn/T/tmp-59344-nWjRYDe81Aoe/vault",
      },
    },
    Object {
      "body": "
",
      "children": Array [],
      "created": "1",
      "custom": Object {},
      "data": Object {},
      "desc": "",
      "fname": "foo.ch1",
      "id": "foo.ch1",
      "links": Array [],
      "parent": "foo",
      "schema": Object {
        "moduleId": "foo",
        "schemaId": "ch1",
      },
      "title": "Ch1",
      "type": "note",
      "updated": "1",
      "vault": Object {
        "fsPath": "/var/folders/65/c9lrx0tx6k7b_zlb4v1bs2rr0000gn/T/tmp-59344-nWjRYDe81Aoe/vault",
      },
    },
    Object {
      "body": "
",
      "children": Array [
        "foo",
      ],
      "created": "1",
      "custom": Object {},
      "data": Object {},
      "desc": "",
      "fname": "root",
      "id": "0a128e8c-bba2-440e-92e8-89a311a17512",
      "links": Array [],
      "parent": null,
      "title": "Root",
      "type": "note",
      "updated": "1",
      "vault": Object {
        "fsPath": "/var/folders/65/c9lrx0tx6k7b_zlb4v1bs2rr0000gn/T/tmp-59344-nWjRYDe81Aoe/vault",
      },
    },
  ],
  "error": null,
}
`;

exports[`notes query query root: root note 1`] = `
Object {
  "data": Array [
    Object {
      "body": "
",
      "children": Array [
        "foo",
      ],
      "created": "1",
      "custom": Object {},
      "data": Object {},
      "desc": "",
      "fname": "root",
      "id": "7afee663-5e8b-46a3-be13-b33c4ca0239e",
      "links": Array [],
      "parent": null,
      "title": "Root",
      "type": "note",
      "updated": "1",
      "vault": Object {
        "fsPath": "/var/folders/65/c9lrx0tx6k7b_zlb4v1bs2rr0000gn/T/tmp-59344-H2ktP2m4j41r/vault",
      },
    },
  ],
  "error": null,
}
`;

exports[`notes rename/ basic 1`] = `
Object {
  "data": Array [
    Object {
      "note": Object {
        "body": "[[baz]]

",
        "children": Array [],
        "created": "1",
        "custom": Object {},
        "data": Object {},
        "desc": "",
        "fname": "bar",
        "id": "bar",
        "links": Array [
          Object {
            "alias": "baz",
            "from": Object {
              "fname": "bar",
              "id": "bar",
            },
            "original": "baz",
            "pos": Object {
              "end": 7,
              "start": 0,
            },
            "to": Object {
              "fname": "baz",
            },
            "type": "wiki",
            "value": "baz",
          },
        ],
        "parent": "56680149-b498-44a1-b0c6-f4c9f2041dae",
        "title": "Bar",
        "type": "note",
        "updated": "1",
        "vault": Object {
          "fsPath": "/var/folders/65/c9lrx0tx6k7b_zlb4v1bs2rr0000gn/T/tmp-59344-TmGJut7ZX9iK/vault",
        },
      },
      "status": "update",
    },
    Object {
      "note": Object {
        "body": "[[bar]]
",
        "children": Array [
          "foo.ch1",
        ],
        "created": "1",
        "custom": Object {},
        "data": Object {},
        "desc": "",
        "fname": "foo",
        "id": "foo",
        "links": Array [],
        "parent": "56680149-b498-44a1-b0c6-f4c9f2041dae",
        "title": "Foo",
        "type": "note",
        "updated": "1",
        "vault": Object {
          "fsPath": "/var/folders/65/c9lrx0tx6k7b_zlb4v1bs2rr0000gn/T/tmp-59344-TmGJut7ZX9iK/vault",
        },
      },
      "status": "delete",
    },
    Object {
      "note": Object {
        "body": "[[bar]]
",
        "children": Array [
          "foo.ch1",
        ],
        "created": "1",
        "custom": Object {},
        "data": Object {},
        "desc": "",
        "fname": "baz",
        "id": "foo",
        "links": Array [],
        "parent": "56680149-b498-44a1-b0c6-f4c9f2041dae",
        "title": "Foo",
        "type": "note",
        "updated": "1",
        "vault": Object {
          "fsPath": "/var/folders/65/c9lrx0tx6k7b_zlb4v1bs2rr0000gn/T/tmp-59344-TmGJut7ZX9iK/vault",
        },
      },
      "status": "create",
    },
  ],
  "error": null,
}
`;

exports[`notes rename/ basic 2`] = `
Array [
  "bar.md",
  "baz.md",
  "foo.ch1.md",
  "foo.schema.yml",
  "root.md",
  "root.schema.yml",
]
`;

exports[`notes rename/ domain with no children, update exsiting node: changed 1`] = `
Array [
  Object {
    "note": Object {
      "body": "[[gamma]]

",
      "children": Array [],
      "created": "1",
      "data": Object {},
      "desc": "",
      "fname": "alpha",
      "id": "alpha",
      "links": Array [
        Object {
          "alias": "gamma",
          "from": Object {
            "fname": "alpha",
            "id": "alpha",
          },
          "original": "gamma",
          "pos": Object {
            "end": 9,
            "start": 0,
          },
          "to": Object {
            "fname": "gamma",
          },
          "type": "wiki",
          "value": "gamma",
        },
      ],
      "parent": "6fd9cc62-5d63-47c8-8172-e9d8ad918de8",
      "title": "Alpha",
      "type": "note",
      "updated": "1",
      "vault": Object {
        "fsPath": "/var/folders/65/c9lrx0tx6k7b_zlb4v1bs2rr0000gn/T/tmp-59344-zXCq9aw0VJ3G/vault",
      },
    },
    "status": "update",
  },
  Object {
    "note": Object {
      "body": "[[alpha]]
",
      "children": Array [],
      "created": "1",
      "custom": Object {},
      "data": Object {},
      "desc": "",
      "fname": "beta",
      "id": "beta",
      "links": Array [],
      "parent": "6fd9cc62-5d63-47c8-8172-e9d8ad918de8",
      "title": "Beta",
      "type": "note",
      "updated": "1",
      "vault": Object {
        "fsPath": "/var/folders/65/c9lrx0tx6k7b_zlb4v1bs2rr0000gn/T/tmp-59344-zXCq9aw0VJ3G/vault",
      },
    },
    "status": "delete",
  },
  Object {
    "note": Object {
      "body": "[[alpha]]
",
      "children": Array [],
      "created": "1",
      "custom": Object {},
      "data": Object {},
      "desc": "",
      "fname": "gamma",
      "id": "beta",
      "links": Array [],
      "parent": "6fd9cc62-5d63-47c8-8172-e9d8ad918de8",
      "title": "Beta",
      "type": "note",
      "updated": "1",
      "vault": Object {
        "fsPath": "/var/folders/65/c9lrx0tx6k7b_zlb4v1bs2rr0000gn/T/tmp-59344-zXCq9aw0VJ3G/vault",
      },
    },
    "status": "create",
  },
]
`;

exports[`schema write simple schema 1`] = `
"version: 1
imports: []
schemas:
  - id: pro
    children: []
    title: pro
    parent: root
"
`;
